includes/class-telemetry.php

```
@@ -0,0 +1,648 @@
+					foreach ( explode( "\r\n", $last_block ) as $line ) {
+						$parts = explode( ':', $line, 2 );
+						if ( 2 === count( $parts ) ) {
+							$headers[ strtolower( trim( $parts[0] ) ) ] = trim( $parts[1] );
+						}
+					}
+
+					// Precise network timings from cURL info struct.
+					$timings   = array(
+						'dns'     => round( $info['namelookup_time'] * 1000, 2 ),
+						'connect' => round( ( $info['connect_time'] - $info['namelookup_time'] ) * 1000, 2 ),
+						'ssl'     => ( isset( $info['appconnect_time'] ) && $info['appconnect_time'] > 0 )
```

PHPStan (2.1.46)

[warning] Offset 'appconnect_time' on (array{url: string, content_type: string|null, http_code: int, header_size: int, request_size: int, filetime: int, ssl_verify_result: int, redirect_count: int, ...}|false) in isset() does not exist.


(isset.offset)